From 5c8e74b76b5dd8f65fb54972f3ce23ebc411c51e Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Fri, 2 Sep 2022 17:21:24 +0200
Subject: [PATCH 02/21] linux: Allow filling only the serial from sibling

This fixes the serial number not being set on Bluetooth devices which
might not have a serial number but should always have a Bluetooth
address to differentiate them.
---
 src/linux/up-device-supply.c | 16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

diff --git a/src/linux/up-device-supply.c b/src/linux/up-device-supply.c
index ab50c67..7cf452a 100644
--- a/src/linux/up-device-supply.c
+++ b/src/linux/up-device-supply.c
@@ -367,19 +367,21 @@ up_device_supply_sibling_discovered (UpDevice *device,
 		      "serial", &serial_number,
 		      NULL);
 
-	if (model_name == NULL && serial_number == NULL) {
+	if (model_name == NULL) {
 		model_name = up_device_supply_get_string (input, "name");
-		serial_number = up_device_supply_get_string (input, "uniq");
-
 		up_make_safe_string (model_name);
-		up_make_safe_string (serial_number);
-
 		g_object_set (device,
 			      "model", model_name,
-			      "serial", serial_number,
 			      NULL);
-
 		g_free (model_name);
+	}
+
+	if (serial_number == NULL) {
+		serial_number = up_device_supply_get_string (input, "uniq");
+		up_make_safe_string (serial_number);
+		g_object_set (device,
+			      "serial", serial_number,
+			      NULL);
 		g_free (serial_number);
 	}
 
-- 
2.39.1

